gflownet/proxy/ising.py

```diff
@@ -0,0 +1,90 @@
+import itertools
+from typing import Optional, Union
+
+import torch
+from torchtyping import TensorType
+
+from gflownet.proxy.base import Proxy
+
+
+def nn_adjacency(
+    n_dim: int, length: int, J_nn: float = 1.0, periodic: bool = True
+) -> torch.Tensor:
+    """
+    Construct adjacency matrix for a hypercubic lattice of size length**n_dim with
+    nearest-neighbor coupling J_nn.
+    """
+    N = length**n_dim
+    J = torch.zeros((N, N), dtype=torch.float)
+
+    coords = list(itertools.product(range(length), repeat=n_dim))
+    coords_to_index = {c: i for i, c in enumerate(coords)}
+
+    for c in coords:
+        i = coords_to_index[c]
+        for axis in range(n_dim):
+            for shift in [-1, 1]:  # For each axis, we consider both directions :
+                # the neighbor on the left (-1) and the neighbor on the right (+1).
+                neighbor = list(c)
+                neighbor[axis] += shift
+
+                if periodic:
+                    neighbor[axis] %= length
+                elif neighbor[axis] < 0 or neighbor[axis] >= length:
+                    continue  # if no periodic conditions, this is not a valid neighbor
+
+                j = coords_to_index[tuple(neighbor)]
+                J[i, j] = J_nn
+    return J
+
+
+class Ising(Proxy):
+    def __init__(
+        self,
+        n_dim: Optional[int] = None,  # No need to provide n_dim if J is provided
```

Comment:
This kind of comment about the parameters should go in the docstring of the method instead of here.